lib/ld-eventsource/client.rb

```diff
@@ -128,8 +129,6 @@ def initialize(uri,
           read: read_timeout,
           connect: connect_timeout
         })
-      @buffer = ""
-      @lock = Mutex.new
```

Comment:
The `@buffer` is now a local variable inside `BufferedLineReader.lines_from`. The `@lock` was never actually necessary, since we were only using it for synchronization within a code path that was always on the same thread.